app/lib/shared/search_service.dart

```diff
@@ -73,17 +73,69 @@ class PackageDocument extends Object with _$PackageDocumentSerializerMixin {
       _$PackageDocumentFromJson(json);
 }
 
+/// How search results should be ordered.
+enum SearchOrder {
+  /// Search score should be a weighted value of [text], [updated],
+  /// [popularity], [health] and [maintenance], ordered decreasing.
+  overall,
+
+  /// Search score should depend only on text match similarity, ordered
+  /// decreasing.
+  text,
+
+  /// Search order should be in decreasing last updated time.
+  updated,
+
+  /// Search order should be in decreasing popularity score.
+  popularity,
+
+  /// Search order should be in decreasing health score.
+  health,
+
+  /// Search order should be in decreasing maintenance score.
+  maintenance,
+}
+
+/// Returns null if [value] is not a recognized search order.
+SearchOrder parseSearchOrder(String value, {SearchOrder orElse()}) {
```

Comment:
Why not just change `{SearchOrder orElse()}` to `{SearchOrder defaultsTo}`? Then call sites don't need to allocate a closure but can just provide a constant.